src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -161,28 +161,39 @@ public IDataReader ExecuteQuery(string query, CancellationToken cancellationToke
             IDataReader[] origReaders = new IDataReader[script.Blocks.Count];
             try
             {
+                var numOfQueries = 0;
                 Parallel.ForEach(script.Blocks, (codeBlock, state, index) =>
                 {
                     var minimalQuery =
                         codeBlock.Service.GetMinimalText(MinimalTextKind.RemoveLeadingWhitespaceAndComments);
                     
-                    IDataReader origReader;
-                
-                    if(minimalQuery.StartsWith(".") && !minimalQuery.StartsWith(".show")){
-                        origReader = _kustoAdminProvider.ExecuteControlCommand(
-                            KustoQueryUtils.IsClusterLevelQuery(minimalQuery) ? "" : databaseName,
-                            minimalQuery,
-                            clientRequestProperties);
```

Comment:
I think we need a new clientRequestProperties for each call to .Execute...(). We should pull that from the top of this function into this loop. Or create helper functions which will initialize clentRequestProperties and call the .Execute...() functions.